code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,712 @@
+ 
+  /edge-cloud-nodes:
+    get:
+      tags: 
+        - Edge Cloud
+      summary: Retrieve a list of the operator’s Edge Clouds and their status
+      description: List of the operator’s Edge Clouds and their status, ordering the results by location and filtering by status (active/inactive/unknown)
+      operationId: getEdgeCloudNodes
+      parameters:
+        - name: regionId
+          description: Human readable name of the geografical zone of the Edge Cloud. Defined by the OP.
+          in: query
+          required: false
+          schema: 
+              $ref: '#/components/schemas/RegionId'
+        - name: status
+          in: query
+          required: false
+          schema: 
+              $ref: '#/components/schemas/Status'
+      responses:
+        '200':
+          description: Available Edge Cloud Nodes 
+          content:
+            application/json:
+              schema:
+                type: array
+                items:                  
+                  $ref: '#/components/schemas/EdgeCloudNodeDetails'
+                minItems: 1
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+ 
+components:
+  securitySchemes:
+    oAuth2ClientCredentials:
+      type: oauth2
+      flows: 
+        clientCredentials: 
+          tokenUrl: '/oauth2/token'
+          scopes:
+            nbi-mgmt: Access to the APIs
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated with the application. OP generates this identifier when the application is submitted over NBI.
+    AppInstanceId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated with a running instance of an application. OP generates this identifier. 
+
+    AppInstantiation:
+      description: Attributes with information where to instantiate a given application
+      type: object
+      properties:
+        appId:
+          $ref: '#/components/schemas/AppId'
+        edgeCloudNodeName:
+          $ref: '#/components/schemas/EdgeCloudNodeName'
+        regionId:
+          $ref: '#/components/schemas/RegionId'
+      required:
+        - appId
+
+    AppManifest:
+      properties:
+        name:
+          type: string
+          description: Name of the application. 
+        version:
+          type: integer
+          description: Application version information
+        virtType:
+          type: string
+          enum:
+            - VM
+            - CONTAINER
+            - HELM 
+        fileName:
+          type: string
+          minLength:  8
+          maxLength: 32
+          description: Name of the file with the extension eg. zip, targz, etc.
```

Comment:
It need to be described what is expected from the app developer from the fileName? Which file this element is referring to?

Reply:
It also needs to be described if the fileName has any relation with the repository attribute below? That means the expectation is that OP will try to pull the file(s) from the repository?

Reply:
The idea we were thinking was that the combination of both fields `FileName` and `url` will let the developer to build the correct path to the binary file in their repository.
Agree this needs to be documented.